buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/shared-configuration.kt

```diff
@@ -90,14 +90,16 @@ fun Project.createTestTask(name: String, executer: String, sourceSet: SourceSet,
         extraConfig.execute(this)
     }
 
-
-fun splitIntoBuckets(sourceFiles: List<File>, numberOfSplits: Int): List<List<File>> =
-    if (sourceFiles.size % numberOfSplits == 0) {
-        sourceFiles.chunked(sourceFiles.size / numberOfSplits)
-    } else {
-        sourceFiles.chunked(sourceFiles.size / numberOfSplits + 1)
+/**
+ * Distributed test requires all dependencies to be declared
+ */
+fun Project.makeIntegrationTestsDependOnSampleDir(vararg sampleDirs: String) {
+    tasks.withType<IntegrationTest>() {
+        inputs.files(sampleDirs.map { rootProject.fileTree(it) }.toTypedArray())
```

Comment:
```suggestion
        inputs.files(rootProject.files(it))
```

No need to use `fileTree`. Using `toTypedArray()` probably also resolves the `fileTree()`, which seems like a bad idea.